TYPO3.Fluid/Classes/TYPO3/Fluid/ViewHelpers/Link/ActionViewHelper.php

```diff
@@ -88,6 +89,13 @@ public function render($action, $arguments = array(), $controller = null, $packa
             }
             $uriBuilder = clone $uriBuilder;
             $uriBuilder->setRequest($request->getParentRequest());
+        } elseif ($useMainRequest === true) {
+            $request = $this->controllerContext->getRequest();
+            if ($request->isMainRequest()) {
```

Comment:
Should we really throw an exception in this case and not just code around to allow that? I think in the useParentRequest case that makes sense, because "parentRequest" could unexpectedly be the main request in that case, but when the user explicitly wants the main request and is already in it - it's safe to just gracefully use the current request.